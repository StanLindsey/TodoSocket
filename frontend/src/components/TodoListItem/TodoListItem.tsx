import React from "react";
import {
  ListItem,
  ListItemIcon,
  Checkbox,
  ListItemText,
  Divider
} from "@material-ui/core";

const TodoListItem = ({
  value,
  onToggle,
  checked,
  labelId
}: {
  value: number;
  onToggle: (value: number) => void;
  checked: boolean;
  labelId: string;
}) => (
  <div>
    <Divider />
    <ListItem role={undefined} dense button onClick={() => onToggle(value)}>
      <ListItemIcon>
        <Checkbox
          edge="start"
          checked={checked}
          tabIndex={-1}
          disableRipple
          inputProps={{ "aria-labelledby": labelId }}
        />
      </ListItemIcon>
      <ListItemText id={labelId} primary={`Line item ${value + 1}`} />
    </ListItem>
  </div>
);

export default TodoListItem;
