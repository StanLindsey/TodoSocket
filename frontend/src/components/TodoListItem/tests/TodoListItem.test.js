import React from "react";
import renderer from "react-test-renderer";
import TodoListItem from "..";

describe("app", () => {
  it("matches default snapshot", () => {
    const props = {
      value: 1,
      onToggle: () => {},
      checked: false,
      labelId: "Hurro World"
    };

    const tree = renderer.create(<TodoListItem />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
