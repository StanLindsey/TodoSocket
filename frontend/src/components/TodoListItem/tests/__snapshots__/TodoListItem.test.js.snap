// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app matches default snapshot 1`] = `
<div>
  <hr
    className="MuiDivider-root"
  />
  <div
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    <div
      className="MuiListItemIcon-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-29 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary MuiIconButton-edgeStart"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            className="PrivateSwitchBase-input-32"
            data-indeterminate={false}
            disabled={false}
            onChange={[Function]}
            tabIndex={-1}
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
      </span>
    </div>
    <div
      className="MuiListItemText-root MuiListItemText-dense"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
      >
        Line item NaN
      </span>
    </div>
  </div>
</div>
`;
